### ISSUE TITLE
Editing an annotation such that it no longer matches filters causes it to disappear

### ISSUE THREAD

**Issue by seanh** [hypothesis/client#10]
![untitled screencast - edited 25](https://cloud.githubusercontent.com/assets/22498/16523510/bf932ab0-3f9b-11e6-974c-6a4f1b7f0b17.gif)

Suggestion: any annotation that matched the search at the time when the search was first made should remain in the search results until the search query is changed, even if edits to the annotation mean that it no longer matches the search.

(Note: you can also see [this separate issue](https://github.com/hypothesis/client/issues/9) in the gif above.)

**Comment by lyzadanger** [comment]
I renamed this issue because we've recently solved the issue with _new_ annotations/replies disappearing on save when a search filter is active. This only applies to editing _existing_ annotations.

The desired behavior here of keeping an edited annotation visible even if it no longer matches filters bears a little more though, IMHO. At the very least, it may need to wait until after we've tackled https://github.com/hypothesis/client/issues/1242 — it will be more feasible if non-matching annotations are clearly delineated from ones that do match the current filters.
